apmpackage/genfields.go

```diff
@@ -0,0 +1,149 @@
+package apmpackage
+
+import (
+	"gopkg.in/yaml.v2"
+	"io/ioutil"
+	"path/filepath"
+)
+
+func GenerateFields(ecsDir string) {
+	// TODO remove all field files before generate them
+
+	// TODO get this from GH directly
+	ecsFlatFields := loadECSFields(ecsDir)
+
+	inputFieldsFiles := map[string][]string{
+		"logs":    {"model/error/_meta/fields.yml"},
+		"metrics": {"model/metricset/_meta/fields.yml", "model/profile/_meta/fields.yml"},
+		"traces":  {"_meta/fields.common.yml", "model/transaction/_meta/fields.yml", "model/span/_meta/fields.yml"},
```

Comment:
Why is the `_meta/fields.common.yml` only used for traces? It contains fields that are also relevant for errors and some of them for metricsets.